src/linear_algebra/basic.lean

```diff
@@ -2257,6 +2257,15 @@ def to_linear_equiv (e : M ≃ M₂) (h : is_linear_map R (e : M → M₂)) : M
 
 end equiv
 
+namespace add_equiv
+variables [semiring R] [add_comm_monoid M] [semimodule R M] [add_comm_monoid M₂] [semimodule R M₂]
+
+/-- An additive equivalence whose underlying function preserves `smul` is a linear equivalence. -/
+def to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) : M ≃ₗ[R] M₂ :=
+{ map_smul' := h, .. e, }
```

Comment:
```suggestion
{ map_smul' := h, .. e, }

@[simp, norm_cast] lemma coe_to_linear_equiv (e : M ≃+ M₂) (h) :
  ⇑(e.to_linear_equiv h) = e :=
rfl

@[simp, norm_cast] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h) :
  ⇑(e.to_linear_equiv h).symm = e.symm :=
rfl
```

Reply:
Isn't the second one implied by the first?

Reply:
Ah, perhaps my understanding of the precedence of the arrow is wrong

Reply:
@urkud, this:
```
@[simp, norm_cast] lemma coe_to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
  ⇑(e.to_linear_equiv h) = e :=
rfl
```
is interpreted as `⇑(e.to_linear_equiv h) = ⇑e`, and gives `norm_cast: badly shaped lemma, rhs can't start with coe`

Reply:
Do you want me to drop the `norm_cast`, as:
```
@[simp] lemma coe_to_linear_equiv (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
  ⇑(e.to_linear_equiv h) = e :=
rfl

@[simp] lemma coe_to_linear_equiv_symm (e : M ≃+ M₂) (h : ∀ (c : R) x, e (c • x) = c • e x) :
  ⇑(e.to_linear_equiv h).symm = e.symm :=
rfl
```

Reply:
Yes, please.

Reply:
Are you sure this is a good simp lemma? It throws away structure, so `⇑(e.to_linear_equiv h) (c • x)` becomes becomes `⇑e (c • x)`, which no longer has enough information to be rewritten to `c • ⇑(e.to_linear_equiv h) x`

Reply:
We simplify, e.g., `⇑(equiv.mk f g h₁ h₂)` to `f`. I don't know if setting `priority` to something less than `1000` will make `⇑(e.to_linear_equiv h) (c • x)` simplify to `c • e x`. This may be a good topic to discuss on Zulip but now I'd expect `mathlib` to have this simp lemma.

Reply:
Either way, I've pushed a change that does as you suggested